Src/Support/Google.Apis.Auth.Tests/OAuth2/ImpersonatedCredentialTests.cs

```diff
@@ -0,0 +1,230 @@
+        {
+            var sourceCredential = CreateSourceCredential();
+            var body = new { token = OidcComputeSuccessMessageHandler.FirstCallToken };
+            var content = NewtonsoftJsonSerializer.Instance.Serialize(body);
+            var messageHandler = new MockHttpMessageHandler(HttpStatusCode.OK, content);
+            var initializer = new ImpersonatedCredential.Initializer(sourceCredential, "foo", new string[] {"scope"})
+            {
+                Clock = clock,
+                HttpClientFactory = new MockHttpClientFactory(messageHandler)
+            };
+            return new ImpersonatedCredential(initializer);
+        }
+
+        private ImpersonatedCredential CreateImpersonatedCredentialWithAccessTokenResponse()
+        {
+            var sourceCredential = CreateSourceCredential();
+            var body = new {
+                accessToken = "access_token",
+                expireTime = "2020-05-13T16:00:00.045123456Z"
+            };
+            var content = NewtonsoftJsonSerializer.Instance.Serialize(body);
+            var messageHandler = new MockHttpMessageHandler(HttpStatusCode.OK, content);
+            var initializer = new ImpersonatedCredential.Initializer(sourceCredential, "foo", new string[] {"scope"})
+            {
+                Clock = clock,
+                HttpClientFactory = new MockHttpClientFactory(messageHandler)
+            };
+            return new ImpersonatedCredential(initializer);
+        }
+
+        private ImpersonatedCredential CreateImpersonatedCredentialWithSignBlobResponse()
+        {
+            var sourceCredential = CreateSourceCredential();
+            // Use signedBlob = base64("foo") = "Zm9v"
+            var body = new { keyId = "1", signedBlob = "Zm9v" };
+            var content = NewtonsoftJsonSerializer.Instance.Serialize(body);
+            var messageHandler = new MockHttpMessageHandler(HttpStatusCode.OK, content);
+            var initializer = new ImpersonatedCredential.Initializer(sourceCredential, "foo", new string[] {"scope"})
+            {
+                Clock = clock,
+                HttpClientFactory = new MockHttpClientFactory(messageHandler)
+            };
+            return new ImpersonatedCredential(initializer);
+        }
+
+        private ImpersonatedCredential CreateImpersonatedCredentialWithErrorResponse()
+        {
+            var sourceCredential = CreateSourceCredential();
+            var content = errorResponseContent;
+            var messageHandler = new MockHttpMessageHandler(HttpStatusCode.NotFound, content);
+            var initializer = new ImpersonatedCredential.Initializer(sourceCredential, "foo", new string[] {"scope"})
+            {
+                Clock = clock,
+                HttpClientFactory = new MockHttpClientFactory(messageHandler)
+            };
+            return new ImpersonatedCredential(initializer);
+        }
+
+        [Fact]
+        public void TestConstructorInvalidSourceCredential()
```

Comment:
Nit: remove the Test prefix from all of these. I'd call this `Constructor_InvalidSourceCredential`.

(The convention is either `MethodBeingTested`, or `MethodBeingTested_Scenario`, or `MethodBeingTested_Scenario_Result` depending on what's needed.)

Reply:
done